### ISSUE TITLE
How to know our accuracy

### ISSUE THREAD

**Issue by Hussain06061997** [ultralytics/yolov5#3533]
## ❔Question

Anyone please help me to give me detail about my confusion matrix performance

![image](https://user-images.githubusercontent.com/67025744/121230708-2bdc4a80-c8c2-11eb-9710-19708ba565a6.png)

**Comment by glenn-jocher** [comment]
@Hussain06061997 your confusion matrix looks great, i.e. good training results. If you'd like to learn more about confusion matrices in general you can visit https://en.wikipedia.org/wiki/Confusion_matrix

**Comment by mansi-aggarwal-2504** [comment]
@glenn-jocher I have a question. The confusion matrix does tell the percentage of correctly labeled objects and suppose I have only one class where it'll give the background FP corresponding to that label as 1.0 i.e. 100% (naturally, since all FP belong to that class).
But I can't compare the FP percentages. For example, one model is mapping 76% objects and second is mapping 81% but the second model is also bringing more false positives. 
**Model 1:**
![confusion_matrix](https://user-images.githubusercontent.com/60866597/121641696-23943300-caad-11eb-9e2b-ff3abdb5e147.png)

**Model 2:** 
![confusion_matrix](https://user-images.githubusercontent.com/60866597/121641734-30188b80-caad-11eb-95ea-99309c454895.png)

I don't get that info from the yolov5 confusion matrix. Is there any way I can get that? 

(my images include 50-100 small objects in a single picture so its difficult to check FPs by manual inspection)

**Comment by glenn-jocher** [comment]
@mansi-aggarwal-2504 you can comment this line for absolute values:
https://github.com/ultralytics/yolov5/blob/5c32bd3080c8643aed9c167bb2fc655f502facaf/utils/metrics.py#L165

**Comment by mansi-aggarwal-2504** [comment]
@glenn-jocher it's working. Thank you!

**Comment by glenn-jocher** [comment]
@mansi-aggarwal-2504 good news 😃! I've updated the code in ✅ in PR #3586 and #3587 to make this easier by providing a `normalize` flag in the ConfisionMatrix plot method. To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by pravastacaraka** [comment]
@glenn-jocher Can you tell me what is background FN and background FP?

**Comment by mansi-aggarwal-2504** [comment]
@pravastacaraka background FN/FP are the percentage of FN/FP that belong to a particular class in your dataset. If you scroll up and see the confusion matrix in my comment, since I have only one class, all FPs are belong to that class and hence, it’s 1.0 (i.e. 100%) and there are 24% flowers that were not detected (FNs).
